sql/distsqlplan: TestDistAggregationTable failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/58c567a325056033b326cb9c4ed9ba490e8956da

Parameters:
```
TAGS=
GOFLAGS=-race -parallel=2
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestDistAggregationTable PKG=github.com/cockroachdb/cockroach/pkg/sql/distsqlplan TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1296842&tab=buildLog

```
I190520 05:35:08.024861 28 server/status/recorder.go:612  [n1] available memory from cgroups (8.0 EiB) exceeds system memory 16 GiB, using system memory
I190520 05:35:08.025017 28 server/server.go:1925  [n1] Could not start heap profiler worker due to: directory to store profiles could not be determined
I190520 05:35:08.027397 28 server/server.go:1592  [n1] starting https server at 127.0.0.1:41611 (use: 127.0.0.1:41611)
I190520 05:35:08.027541 28 server/server.go:1594  [n1] starting grpc/postgres server at 127.0.0.1:35577
I190520 05:35:08.027667 28 server/server.go:1595  [n1] advertising CockroachDB node at 127.0.0.1:35577
I190520 05:35:08.059471 145 util/protoutil/randnullability.go:94  inserting null for (roachpb.Lease).Expiration: false
I190520 05:35:08.059571 145 util/protoutil/randnullability.go:94  inserting null for (roachpb.Lease).DeprecatedStartStasis: false
I190520 05:35:08.164700 322 sql/event_log.go:135  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I190520 05:35:08.239155 268 sql/event_log.go:135  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:19.1-2 User:root}
I190520 05:35:08.280451 466 sql/event_log.go:135  [n1,intExec=disableNetTrace] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:root}
I190520 05:35:08.379029 337 sql/event_log.go:135  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:59053235-4279-44fe-b659-10947bcc2d3c User:root}
I190520 05:35:08.417422 645 sql/event_log.go:135  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I190520 05:35:08.440295 598 sql/event_log.go:135  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I190520 05:35:08.477567 28 server/server.go:1663  [n1] done ensuring all necessary migrations have run
I190520 05:35:08.477722 28 server/server.go:1666  [n1] serving sql connections
I190520 05:35:08.492289 603 server/server_update.go:72  [n1] no need to upgrade, cluster already at the newest version
I190520 05:35:08.494557 606 sql/event_log.go:135  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:35577 Attrs: Locality:region=test,dc=dc1 ServerVersion:19.1-2 BuildTag:v19.2.0-alpha.00000000-489-g58c567a StartedAt:1558330507992828765 LocalityAddress:[] XXX_NoUnkeyedLiteral:{} XXX_sizecache:0} ClusterID:e12c9a59-e6ae-4dfd-9654-819fafbd2d66 StartedAt:1558330507992828765 LastUp:1558330507992828765}
I190520 05:35:08.564855 822 sql/event_log.go:135  [n1,client=127.0.0.1:38572,user=root] Event: "create_database", target: 52, info: {DatabaseName:test Statement:CREATE DATABASE IF NOT EXISTS test User:root}
I190520 05:35:08.590117 822 sql/event_log.go:135  [n1,client=127.0.0.1:38572,user=root] Event: "create_table", target: 53, info: {TableName:test.public.t Statement:CREATE TABLE test.public.t (k INT8 PRIMARY KEY, int1 INT8, int2 INT8, bool1 BOOL, bool2 BOOL, dec1 DECIMAL, dec2 DECIMAL, float1 FLOAT8, float2 FLOAT8, b BYTES) User:root}
I190520 05:35:08.604047 902 storage/replica_command.go:252  [n1,split,s1,r20/1:/{Table/24-Max}] initiating a split of this range at key /Table/53 [r21] (zone config)
--- FAIL: TestDistAggregationTable (13.33s)
=== RUN   TestDistAggregationTable/AVG/float1/10
    --- FAIL: TestDistAggregationTable/AVG/float1/10 (0.01s)
        aggregator_funcs_test.go:375: different results for column 0
            w/o local stage:   0.000
            with local stage:  0.000
=== RUN   TestDistAggregationTable/SUM/float1/10
    --- FAIL: TestDistAggregationTable/SUM/float1/10 (0.01s)
        aggregator_funcs_test.go:375: different results for column 0
            w/o local stage:   0.002
            with local stage:  0.002

```

--- a/pkg/sql/distsqlplan/aggregator_funcs_test.go
+++ b/pkg/sql/distsqlplan/aggregator_funcs_test.go
@@ -341,28 +341,27 @@ func checkDistAggregationInfo(
 					equiv = decNonDist.Coeff.Cmp(bigOne) == 0
 				}
 			case *tree.DFloat:
-				// Float results are highly variable and
-				// loss of precision between non-local and
-				// local is expected. We reduce the precision
-				// specified by floatPrecFmt and compare
-				// their string representations.
+				// Float results are highly variable and loss
+				// of precision between non-local and local is
+				// expected. We reduce the precision specified
+				// by floatPrecFmt and compare their string
+				// representations.
 				floatDist := float64(*typedDist)
 				floatNonDist := float64(*rowNonDist.Datum.(*tree.DFloat))
 				strDist = fmt.Sprintf(floatPrecFmt, floatDist)
 				strNonDist = fmt.Sprintf(floatPrecFmt, floatNonDist)
 				// Compare using a relative equality
-				// func that isn't dependent on the
-				// scale of the number. In addition,
-				// ignore any NaNs. Sometimes due to the
-				// non-deterministic ordering of distsql,
-				// we get a +Inf and Nan result. Both of
-				// these changes started happening with the
-				// float rand datums were taught about some
-				// more adversarial inputs. Since floats
-				// by nature have equality problems and
-				// I think our algorithms are correct,
-				// we need to be slightly more lenient in
-				// our float comparisons.
+				// func that isn't dependent on the scale
+				// of the number. In addition, ignore any
+				// NaNs. Sometimes due to the non-deterministic
+				// ordering of distsql, we get a +Inf and
+				// Nan result. Both of these changes started
+				// happening with the float rand datums
+				// were taught about some more adversarial
+				// inputs. Since floats by nature have equality
+				// problems and I think our algorithms are
+				// correct, we need to be slightly more lenient
+				// in our float comparisons.
 				equiv = almostEqualRelative(floatDist, floatNonDist) || math.IsNaN(floatNonDist) || math.IsNaN(floatDist)
 			default:
 				// For all other types, a simple string
@@ -396,7 +395,7 @@ func almostEqualRelative(a, b float64) bool {
 	if B > A {
 		largest = B
 	}
-	const maxRelDiff = 1e-13
+	const maxRelDiff = 1e-10
 	return diff <= largest*maxRelDiff
 }
 

